plugins/com.google.cloud.tools.eclipse.appengine.libraries.ui/src/com/google/cloud/tools/eclipse/appengine/libraries/ui/CloudLibrariesPage.java

```diff
@@ -101,23 +106,33 @@ public void initialize(IJavaProject project, IClasspathEntry[] currentEntries) {
     if (libraries == null || libraries.isEmpty()) {
       return null;
     }
-  
+
     try {
       if (MavenUtils.hasMavenNature(project.getProject())) {
         BuildPath.addMavenLibraries(project.getProject(), libraries, new NullProgressMonitor());
         return new IClasspathEntry[0];
       } else {
+        Set<LibraryFile> masterFiles = new HashSet<>();
         for (Library library : libraries) {
           if (!library.isResolved()) {
             library.resolveDependencies();
           }
+          masterFiles.addAll(library.getLibraryFiles());
         }
+
+        Library masterLibrary = CloudLibraries.getMasterLibrary();
+        masterFiles.addAll(masterLibrary.getLibraryFiles());
+        masterLibrary.setLibraryFiles(masterFiles);
+        ArrayList<Library> masterLibraries = new ArrayList<>();
+        masterLibraries.add(masterLibrary);
+        
+        // todo This takes a long time. Use a real progress monitor
         IClasspathEntry[] added =
-            BuildPath.listAdditionalLibraries(project, libraries, new NullProgressMonitor());
+            BuildPath.addNativeLibraries(project, masterLibraries, new NullProgressMonitor());
```

Comment:
We should keep `list..` here, otherwise we will hit #1558 we fixed a long time ago. #1558 doesn't happen for App Engine projects created by our wizards, but this happens when you try to add libraries on a plain Java project.

Reply:
However, I noticed one more problem even if we kept the `list..()`: it won't update the container the second time a user tries to add new libraries. This is because `runContainerResolverJob()` is not called in `BuildPath.prepareLibraries()` if `addToClasspath` is `false`. It is the `runContainerResolverJob()` method that can actually update the newly set library files.

Reply:
Restored list, modulo some refactoring. Still need to figure out what to do about the runContainerResolverJob()

Reply:
Now runs the container resolver job after listing